From cfe12915e4edb380254d6f79feb34ee0f92dbbe2 Mon Sep 17 00:00:00 2001
From: "Gary E. Miller" <gem@rellim.com>
Date: Thu, 17 May 2012 11:17:51 -0700
Subject: [PATCH] dvancing the twirly baton only each 11th line, the whole
 regress-driver run is sped up by 7% when run over SSH.
 Patch by: Beat Bolli <bbolli@ewanet.ch>

---
 gpsfake |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/gpsfake
+++ b/gpsfake
@@ -31,10 +31,11 @@ class Baton:
         if os.isatty(self.stream.fileno()):
             if ch:
                 self.stream.write(ch)
-            else:
+                self.stream.flush()
+            elif self.count % 11 == 0:
                 self.stream.write("-/|\\"[self.count % 4])
                 self.stream.write("\010")
-            self.stream.flush()
+                self.stream.flush()
         self.count = self.count + 1
         return
 
